### PR TITLE
docs: add dev version

### PR DESCRIPTION
**Pull request by JacobCoffee**
### Pull Request Checklist

[//]: # "By submitting this issue, you agree to:"
[//]: # "- follow Litestar's [Code of Conduct](https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md)"
[//]: # "- follow Litestar's [Contribution Guidelines](https://litestar.dev/community/contribution-guide)"

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR

### Description
[//]: # "Please describe your pull request for new release changelog purposes"
[//]: # "Example: 'This pr adds the capability to use server-sent events, and documents it in the usage docs'"

-This aims to add a `dev` version to the docs to represent the current state of `main`
We had this before:
image
but we stopped building them
image

### Close Issue(s)
[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

-

### CODE REVIEW

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -29,7 +32,12 @@ jobs:
       - name: Fetch gh pages
         run: git fetch origin gh-pages --depth=1
       - name: Build docs
-        run: poetry run python tools/build_docs.py docs-build
+        run: |
```
Comment (provinzkraut) [inline .github/workflows/docs.yaml:35]: I'd make this into a separate step. Then it's easier to see from within the CI what's going on. So you'd have a `Build release docs` and a `Build dev docs` step

**Comment by JacobCoffee** [comment]
@provinzkraut like so?

**Review by provinzkraut — APPROVED** [review]
Cool! Can you verify on your fork that it works correctly?

**Comment by JacobCoffee** [comment]
It skipped :( 
https://github.com/JacobCoffee/litestar/actions/runs/5629656008/job/15254902385

**Comment by JacobCoffee** [comment]
Seems to be working https://github.com/JacobCoffee/litestar/actions/runs/5629734748/job/15255023488

**Pull request merged**